### PR TITLE
adds SWC Instructor Training and Shell Zenodo citations

### PR DESCRIPTION
**Pull request by raynamharris**
Add reference for https://github.com/gvwilson/collaborative-lesson-development/issues/66

There are some issues with two peoples names with unusual characters...

### CODE REVIEW

File: paper.bib [hunk 1]
```diff
@@ -43,3 +43,34 @@ @book{wiggins-mctighe
   edition = {2nd},
   isbn = {978-1416600350}
 }
+
+@book{Ahmadia2016SoftwareTraining,
+    title = {{Software Carpentry: Instructor Training}},
+    year = {2016},
+    author = {Ahmadia, Aron and Koch, Christina and Wilson, Greg and Allen, James and Banaszkiewicz, Piotr and Becker, Erin and Bekolay, Trevor and Blischak, John and Boughton, Andy and Bray, Erik and Cabunoc Mayes, Abigail and Crouch, Steve and Davis, Neal and Davis, Matt and Duckles, Jonah and Emonet, R\'{e}mi and Fortin, F\'{e}lix-Antoine and Gonzalez, Ivan and Hamm, Chris and Hansen, Michael and Harris, Rayna and Henninger, Felix and Hinsen, Konrad and Hodge, Amy and Jackson, Mike and King, W. Trevor and Kitzes, Justin and Koch, Christina and Liversidge, Tom and Michonneau, Francois and Mills, Bill and Nederbragt, Lex and O'Leary, Aaron and Patitsas, Elizabeth and Pawlik, Aleksandra and Perez, Fernando and Pipitone, Jon and Poisot, Timoth\'{e}e and Rokem, Ariel and Silva, Raniere and Teal, Tracy and Tr��ndle, Tim and Tweedie, Fiona and Vie, Jill-J��nn and Walker, Jordan and Walsh, Alistair and Weaver, Belinda and White, Ethan and Wilkerson, Chandler and Williams, Jason and Wilson, Greg and Van Der Walt, Anelda},
```
Comment (gvwilson) [inline paper.bib]: This loos like everyone who contributed to the style repo as well as everyone who contributed to the content for the instructor training...

Comment (twitwi) [inline paper.bib]: I just opened https://github.com/swcarpentry/swc-releases/issues/4 as it probably needs more discussion about what is a contributor.

For the paper, what do we want? Anyone who has a commit that is in the repo but not also in the style repo?

**Comment by twitwi** [comment]
Related to #66 , I think this would make more sense to cite the most recent version.

**Comment by twitwi** [comment]
@raynamharris did you get the bibtex from zenodo or did you compose it by yourself?
I'm asking because, I have an issue with swcarpentry/swc-releases#4

- Zenodo uses `@misc` by default, which does not include "editors", so we only get the authors in the citation,
- using `@book` solves the issue but is "book" the right type to use for this kind of material

**Comment by twitwi** [comment]
@gvwilson @raynamharris As I'll be unreachable for some time (probably starting Monday), I'll to provide some material in this issue and let you decide and do the integration of it.

**Comment by twitwi** [comment]
For the release 2017.02, compared to the released list of contributors, if we remove the ones only contributing to `styles`, we have:

For shell-novice:
RM:Abigail Cabunoc Mayes
RM:Evan P. Williamson
RM:Jon Pipitone
RM:Michael Hansen

For instructor-training:
RM:Abigail Cabunoc Mayes
RM:Gabriel A. Devenyi
RM:Ian Carroll
RM:Jon Pipitone
RM:Maxim Belkin
RM:Michael Hansen

Then, using `@book` we get:

~~~
@book{devenyi_2017_278226,
  author = {Aldazabal Mensa, Inigo and Alexander, Harriet and Allen, James and Alsheikh-Hussain, Areej and Attali, Dean and Baird, Daniel and Baltzell, Asher and Banaszkiewicz, Piotr and Barmby, Pauline and Beagrie, Rob and Beane, Glen and Bedini, Andrea and Bekolay, Trevor and Belikov, Evgenij and Bell, Jason and Bill, Brian and Blin, Kai and Blischak, John and Boardman, Simon and Boissonneault, Maxime and Bonnie, Jessica and Boughton, Andy and Brase, Ry4an and Brown, Amy and Brunson, Dana and Bryk, Jarek and Buske, Orion and Carlise, Mike and Chen, Daniel and Chung, Kally and Chung, Kathy and Deveau, Daniel and Devenyi, Gabriel A. and Doehle, Phillip and Dolson, Emily and Doyle, Maria and Duckles, Jonah and Emonet, Rémi and Eyers, David and Fernandes, Filipe and Fontenelle, Hugues and Gacenga, Francis and Gardner, Paul and Gidden, Matthew and Gonzalez, Ivan and Gray, Norman and Guillou, Stéphane and Hagh, Varda F. and Harstad, Emelie and Hohenstein, Tom and Howe, Adina and Imamoglu, Fatma and Irving, Damien and Jackson, Mike and Jane McTavish, Emily and Jennings, Michael and Jones, Dan and Jones, Megan A. and Keener, Alix and Keipert, Kristopher and Kelly, Tom and Kim, Jan T. and King, W. Trevor and Koch, Christina and Konrad, Bernhard and Lake, Sherry and Lang, Dustin and Latornell, Doug and Lijnzaad, Philip and Ma, Eric and Mac, Alex and Madin, Joshua and Magle, Tobin and Mandel, Mark and Marini, Camille and Marwaha, Kunal and Mashchenko, Sergey and McCloy, Daniel and Michonneau, François and Middleton, Ryan and Milhans, Jackie and Mills, Bill and Miotto, Amanda and Mount, Sarah and Nederbragt, Lex and Nielsen, Daiva and Nielsen, Danielle and Noga, Klemens and O'Leary, Aaron and Olson, Randy and Orr, Adam and Overgard Therkildsen, Nina and Palamartchouk, Kirill and Perry, Adam and Peterson, Matthew and Poisot, Timothée and Pourreza, Hossein and Povall, Timothy and Richie-Halford, Adam and Ritchie, Scott and Rocha, Cesar B. and Ross, Noam and Rydbeck, Halfdan and Sadjadi, Mahdi and Sanders, Shane and Schloss, Pat and Seyffert, Bertie and Shattow, Genevieve and Silva, Raniere and Simpkin, Sarah and Simpson, John and Smith, Byron and Sobhani, Armin and Soranzo, Nicola and Srinath, Ashwin and Standage, Daniel and Staton, Meg and Steinbach, Peter and Stimberg, Marcel and Telenczuk, Bartosz and Thoele, Florian and Timbers, Tiffany and Turner, Stephen and van der Walt, Anelda and van Schyndel, Jay and Vollmer, David and von der Linden, Jens and Walker, Andrew and Waterfall, Josh and Watson, Greg and White, Ethan and Willing, Carol and Wilson, Greg and Winston, Donny and Wolff, Holger and Young, Lynn and Zamparo, Lee},
  editor = {(Devenyi, Gabriel and Srinath, Ashwin)},
  title = {Software Carpentry: The Unix Shell},
  month = feb,
  year = 2017,
  doi = {10.5281/zenodo.278226},
  url = {https://doi.org/10.5281/zenodo.278226}
}
@book{koch_2017_278229,
  author = {Ahmadiaöê, Aron and Allen, James and Banaszkiewicz, Piotr and Becker, Erin and Bekolay, Trevor and Blischak, John and Boughton, Andy and Bray, Erik and Crouch, Steve and Davis, Matt and Davis, Neal and Duckles, Jonah and Emonet, Rémi and Fortin, Félix-Antoine and Gonzalez, Ivan and Hamm, Chris and Harris, Rayna M. and Henninger, Felix and Hertweck, Kate and Hinsen, Konrad and Hodge, Amy and Jackson, Mike and King, W. Trevor and Kitzes, Justin and Koch, Christina and Lang, Dustin and Liversidge, Tom and Michonneau, François and Mills, Bill and Nederbragt, Lex and Nelson, Ethan and O'Leary, Aaron and Ortiz-Zuazaga, Humberto and Patitsas, Elizabeth and Pawlik, Aleksandra and Perez, Fernando and Poisot, Timothée and Rokem, Ariel and Silva, Raniere and Steyn, Juan and Teal, Tracy and Tröndle, Tim and Tweedie, Fiona and van der Walt, Anelda and Vie, Jill-Jênn and Walker, Jordan and Walsh, Alistair and Weaver, Belinda and White, Ethan and Wilkerson, Chandler and Williams, Jason and Williamson, Evan P. and Wilson, Greg and Woo, Kara},
  editor = {(Koch, Christina and Wilson, Greg)},
  title = {Software Carpentry: Instructor Training},
  month = feb,
  year = 2017,
  doi = {10.5281/zenodo.278229},
  url = {https://doi.org/10.5281/zenodo.278229}
}
~~~


The result in the paper.pdf for plosone merges the numerous authors as "et al.". As we sort our authors alphabetically, we give more credit to A... people. The editors appear. We get this:

1. Aldazabal Mensa I, Alexander H, Allen J, Alsheikh-Hussain A, Attali D, Baird D,
et al. Software Carpentry: The Unix Shell. (Devenyi G, Srinath A, editors; 2017.
Available from: https://doi.org/10.5281/zenodo.278226.
2. Ahmadia A, Allen J, Banaszkiewicz P, Becker E, Bekolay T, Blischak J, et al.
Software Carpentry: Instructor Training. (Koch C, Wilson G, editors; 2017.
Available from: https://doi.org/10.5281/zenodo.278229.

**Comment by twitwi** [comment]
NB: the accents, é, ê, ö seem to be processed ok by my latex.

**Comment by twitwi** [comment]
You might want to remove my name from the authors as I might have contributed somewhat artificially (but not only through styles) to the instructor-training repository. I don't remember if I had real contributions to it.

**Comment by twitwi** [comment]
Linking to https://github.com/swcarpentry/swc-releases/issues/4 as we might want to consolidate what makes most sense.

File: paper.bib (hunk 1 above)
Comment (gvwilson) [inline paper.bib]: I don't think that works - I've committed to both repos, so (committers to this) - (committers to styles) would subtract me.

**Comment by raynamharris** [comment]
@twitwi I imported the citation from Zenodo into the Mendeley reference manager. Then, I hand-edited it to include the editors, which I was able to do by calling it a book. Then, I took the biblitex output from Mendele.  

Does that make sense?

File: paper.bib (hunk 1 above)
Comment (twitwi) [inline paper.bib]: @gvwilson my formulation was ambiguous, there are two ways of seeing this:
- `allAuthors(allCommits(this)) - allAuthors(allCommits(styles))`
- `allAuthors(allCommits(this) - allCommits(styles))`

below (in #71), I used the second one, which removes only a few people in practice.

**Comment by twitwi** [comment]
@raynamharris that makes sense, I was curious of whether Zenodo would propose an option to change the type of publication only at export time.

I still don't know if `@book` is the best semantic, but at least it appears as we want in the citation so this should be ok for now.

**Pull request merged**